parser/src/test/java/io/specmesh/apiparser/AsyncApiParserTest.java

```diff
@@ -116,4 +121,75 @@ public void shouldReturnProducerMessageSchema() {
                 channelsMap.values().iterator().next().publish().message().tags(),
                 Matchers.hasSize(2));
     }
+
+    @Test
+    public void shouldSupportOldSchemaRefLookup() {
+        final var channels = API_SPEC_KEY_VALUE.channels();
+
+        final var oldStyleChannel = channels.get("simple.schema-demo._public.example_old");
+        oldStyleChannel.publish().message().schemaRef();
+
+        try {
+            oldStyleChannel.publish().message().keyType();
+            fail("should have failed on KeyLookup");
+        } catch (Exception ex) {
+
+        }
+        // throw not supports
+        try {
+            oldStyleChannel.publish().message().valueType();
+            fail("should have failed on ValueLookup");
+        } catch (Exception ex) {
+        }
+    }
+
+    @Test
+    public void shouldSupportPrimitiveKeyAndValue() {
+        final var channels = API_SPEC_KEY_VALUE.channels();
+
+        final var oldStyleChannel = channels.get("simple.schema-demo._public.example_primitive");
+
+        try {
+            oldStyleChannel.publish().message().schemaRef();
+            fail("should have failed on SchemaRefLookup");
+        } catch (Exception ex) {
+        }
+
+        final var primitiveKeyType = oldStyleChannel.publish().message().keyType();
+        assertThat(primitiveKeyType, is("string"));
+        final var primitiveValueType = oldStyleChannel.publish().message().valueType();
+        assertThat(primitiveValueType, is("double"));
+    }
+
+    @Test
+    public void shouldSupportSchemaKeyAndValue() {
+        final var channels = API_SPEC_KEY_VALUE.channels();
+
+        final var channel = channels.get("simple.schema-demo._public.example_kv_schema");
+
+        try {
+            channel.publish().message().schemaRef();
```

Comment:
Why couldn't this return "MessageValue.avsc"?

Reply:
mainly because the Message value sits under the payload -> value -> $ref, whereas this is directly under the payload -> $ref - i can make it default - but it will then magically work and could miss picking up the changes correctly